javascript/javascript3/week3/TASK3-Shopping cart/index.js

```diff
@@ -1,130 +1,149 @@
+        // create a temporary new array of products
+        const tempProducts = this.products.map((product) => product);
+        // remove item from temporary array
+        tempProducts.splice(indexOfProduct, 1);
+        // replace products with the temporary array
+        this.products = tempProducts;
+      } else {
+        console.log("Can't find item.");
+      }
+    }
 
-  removeProduct(product) {
-    const indexOfProduct = this.products.indexOf(product);
-    if (indexOfProduct > -1) {
-      // create a temporary new array of products
-      const tempProducts = this.products.map((product) => product);
-      // remove item from temporary array
-      tempProducts.splice(indexOfProduct, 1);
-      // replace products with the temporary array
-      this.products = tempProducts;
-    } else {
-      console.log("Can't find item.");
+    searchProduct(productName) {
+      return this.products.filter((product) =>
+        product.name.toLowerCase().includes(productName.toLowerCase()),
+      );
     }
-  }
 
-  searchProduct(productName) {
-    return this.products.filter((item) => item.name === productName);
-  }
+    getTotal() {
+      return this.products.reduce((result, currentProduct) => result + currentProduct.price, 0);
+    }
 
-  getTotal() {
-    let total = 0;
-    this.products.forEach((item) => {
-      total = item.price + total;
-    });
-    return total;
-  }
+    renderProducts() {
+      const fragment = new DocumentFragment();
+      const ul = document.createElement("UL");
+      fragment.appendChild(ul);
+      this.products.forEach((product) => {
+        createListElement(product);
+      });
+
+      function createListElement(productToList) {
+        let mainLI = document.createElement("LI");
+        ul.appendChild(mainLI);
+        let productUL = document.createElement("UL");
+        mainLI.appendChild(productUL);
+        let nameLI = document.createElement("LI");
+        nameLI.textContent = `Procuct: ${productToList.name}`;
+        productUL.append(nameLI);
+        let priceLI = document.createElement("LI");
+        priceLI.textContent = `Price: ${productToList.price}`;
+        productUL.append(priceLI);
+        document.body.appendChild(fragment);
+      }
+    }
 
-  renderProducts() {
-    const ul = document.createElement("UL");
-    document.body.appendChild(ul);
-    this.products.forEach((product) => {
-      let mainLI = document.createElement("LI");
-      ul.appendChild(mainLI);
-      let productUL = document.createElement("UL");
-      mainLI.appendChild(productUL);
-      let nameLI = document.createElement("LI");
-      nameLI.innerHTML = `Procuct: ${product.name}`;
-      productUL.append(nameLI);
-      let priceLI = document.createElement("LI");
-      priceLI.innerHTML = `Price: ${product.price}`;
-      productUL.append(priceLI);
-    });
+    getUser = async (userID) => {
+      try {
+        const baseUrl = "https://jsonplaceholder.typicode.com/users/";
```

Comment:
instead of passing `baseUrl` around, you can define it as a member of the class and use it from multiple methods. And in TypeScript it would be `private`.